code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeUtils.java

```diff
@@ -175,4 +179,164 @@ static String getRequestEventId(final Event event) {
         }
         return requestEventId;
     }
+
+    /**
+     * Generates a map containing XDM formatted data for {@code Experience Event -
+     * OptimizeProposition Interactions} field group from this {@code OptimizeProposition} offer and
+     * given {@code experienceEventType}.
+     *
+     * <p>The method returns null if the proposition reference within the offer is released and no
+     * longer valid.
+     *
+     * @param experienceEventType {@link String} containing the event type for the Experience Event
+     * @return {@code Map<String, Object>} containing the XDM data for the proposition interaction.
+     */
+    static Map<String, Object> generateInteractionXdm(
+            final List<OptimizeProposition> propositions, final String experienceEventType) {
+
+        final List<Map<String, Object>> decisioningPropositions = new ArrayList<>();
+
+        // Handle the case where propositions list is provided
+        if (propositions != null) {
+            for (OptimizeProposition prop : propositions) {
+                final Map<String, Object> propositionItem = new HashMap<>();
+                final List<Map<String, Object>> propositionItemsList = new ArrayList<>();
+                decisioningPropositions.add(
+                        new HashMap<String, Object>() {
+                            {
+                                put(
+                                        OptimizeConstants.JsonKeys.DECISIONING_PROPOSITIONS_ID,
+                                        prop.getId());
+                                put(
+                                        OptimizeConstants.JsonKeys.DECISIONING_PROPOSITIONS_SCOPE,
+                                        prop.getScope());
+                                put(
+                                        OptimizeConstants.JsonKeys
+                                                .DECISIONING_PROPOSITIONS_SCOPEDETAILS,
+                                        prop.getScopeDetails());
+                                if (!prop.getOffers().isEmpty()
+                                        && prop.getOffers().get(0).getId() != null) {
```

Comment:
Are we supposed to add the item id here for batched display calls? Could you please check if web SDK is doing this?

Reply:
Web team is not sending the items in case of multiple propositions, but sending extra parameter won't harm right ?

Reply:
Accommodated the changes @spoorthipujariadobe